provider/server/exec.go

```diff
@@ -105,17 +104,22 @@ func (s *Server) ServeExec(w http.ResponseWriter, req *http.Request, token strin
 }
 
 func (e executorExec) Exec(ctx context.Context, in io.Reader, out io.WriteCloser, _ remotecommand.TerminalSizeQueue) error {
-	var wg sync.WaitGroup
-
 	machineID := e.ExecRequest.MachineId
-	log := logr.FromContextOrDiscard(ctx).WithName(machineID)
+
+	// Check if a console is already active for this machine
+	if _, ok := activeConsoles.Load(machineID); ok {
```

Comment:
it doesn't look very safe. Combine Load and Store can be little unpredictable, if 2 goroutines run at the same time. I recommend use method LoadOrStore. It will be more safe

Reply:
You're right. Using Load followed by Store is unsafe in a concurrent environment. When two goroutines run simultaneously, both might pass the Load check and then both could try to Store the same machineID, causing a race condition. I'll switch to using the LoadOrStore method, which atomically loads the value for a key if it exists, or stores a new value if the key does not exist. This ensures that only one goroutine will successfully store the machineID as active. Thanks for pointing that out.

Reply:
done